auth/src/main/java/org/aerogear/auth/credentials/IntegrityCheckParameters.java

```diff
@@ -0,0 +1,49 @@
+package org.aerogear.auth.credentials;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+public class IntegrityCheckParameters implements IIntegrityCheckParameters {
+
+    private String issuer;
+    private String audience;
+    private String publicKey;
+
+    public IntegrityCheckParameters(final String audience, final String issuer, final String publicKey) {
+        this.issuer = issuer;
+        this.audience = audience;
+        this.publicKey = publicKey;
+    }
+
+    public IntegrityCheckParameters() {}
+
+    public String getIssuer() {
+        return this.issuer;
+    }
+
+    public String getAudience() {
+        return this.audience;
+    }
+
+    public String getPublicKey() { return this.publicKey; }
```

Comment:
What is this public key? What is the format it is returned? It is a DSA? RSA? or what?